src/Gui/Tree.cpp

```diff
@@ -540,6 +540,31 @@ void TreeWidgetItemDelegate::initStyleOption(QStyleOptionViewItem *option,
     }
 }
 
+class DynamicQLineEdit : public ExpLineEdit
+{
+public:
+    DynamicQLineEdit(QWidget *parent = nullptr) : ExpLineEdit(parent) {}
+
+    QSize sizeHint() const override
+    {
+        QSize size = QLineEdit::sizeHint(); 
+        QFontMetrics fm(font());
+        int availableWidth = parentWidget()->width() - geometry().x() - 20; // Calculate available width
+        size.setWidth(std::min(fm.horizontalAdvance(text()) + 30, availableWidth)); // Add some padding and max width
+        Base::Console().Warning("sizeHint %d %d\n", size.width(), size.height());

```

Comment:
Shouldn't this be a `Log` instead of a `Warning`?

Reply:
There probably shouldn't be any log at all actually, I'll remove it.